entity-framework/core/querying/complex-query-operators.md

```diff
@@ -0,0 +1,120 @@
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#Join)]
+
+```SQL
+SELECT [p].[PersonId], [p].[Name], [p].[PhotoId], [p0].[PersonPhotoId], [p0].[Caption], [p0].[Photo]
+FROM [PersonPhoto] AS [p0]
+INNER JOIN [Person] AS [p] ON [p0].[PersonPhotoId] = [p].[PhotoId]
+```
+
+## GroupJoin
+
+LINQ GroupJoin operator allows you to connect two data sources in same way as Join but the different here is, it creates a group of values from inner source which matches key with element from outer source. Performing query like following generates result of `Blog` & `IEnumerable<Post>`. Since databases (especially relational databases) do not support representing a collection of client side objects, GroupJoin cannot be translated to server in many cases. It requires to get all the data from server to perform GroupJoin without special selector. But if the selector is limiting data being selected then fetching all data from server may cause performance issues. Hence EF Core does not translate GroupJoin.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoin)]
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#GroupJoinComposed)]
+
+## SelectMany
+
+LINQ SelectMany operator allows you to enumerate over collection selector for each element in outer source and generate tuples of value from each data source. In a way it is a join but without any condition so every record is connected with other. Depending on how collection selector is related to outer data source, SelectMany can translate to various different queries on server side.
+
+### Collection selector does not reference outer
+
+When collection selector is not referencing anything from outer source, it is cartesian product of both data sources. In translates to `CROSS JOIN` in relational databases.
+
+[!code-csharp[Main](../../../samples/core/Querying/ComplexQuery/Sample.cs#SelectManyConvertedToCrossJoin)]
+
+```SQL
+SELECT [b].[BlogId], [b].[OwnerId], [b].[Rating], [b].[Url], [p].[PostId], [p].[AuthorId], [p].[BlogId], [p].[Content], [p].[Rating], [p].[Title]
+FROM [Blogs] AS [b]
+CROSS JOIN [Posts] AS [p]
+```
+
+### Collection selector references outer in predicate
+
+When collection selector has a where predicate which is using element from outer source, normally this arises when using collection navigation on outer source as collection selector, then such where predicate can be used as join condition and it can translate to join on databases. If the collection is empty for outer element, then it would not generate any result for that outer element. But if `DefaultIfEmpty` is applied on collection selector then outer element will be connected with default value of inner element. Due to this distinction, this kind of queries translate to `INNER JOIN` in the absence of `DefaultIfEmpty` and `LEFT JOIN` when `DefaultIfEmpty` is applied.
```

Comment:
```suggestion
When collection selector has a where predicate which is using an element from the outer source (normally this arises when using collection navigation on outer source as collection selector) then such where predicate can be used as the join condition and it can be translated to a join on the server side. If the collection is empty for the outer element, then it would not generate any result for that outer element. But if `DefaultIfEmpty` is applied on collection selector then outer element will be connected with the default value of inner element. Due to this distinction, this kind of queries translate to `INNER JOIN` in the absence of `DefaultIfEmpty` and `LEFT JOIN` when `DefaultIfEmpty` is applied.
```